### PR TITLE
Add and improve logs in snapshot scheduling

### PR DESCRIPTION
**Pull request by stephankruggg**
### Description

This PR adds and improves logs in the snapshot scheduling process in order to facilitate troubleshooting, as current logs are scarce.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

### How Has This Been Tested?

I applied the changes in a local lab and checked if the messages were being correctly logged.

### CODE REVIEW

**Comment by harikrishna-patnala** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/storage/snapshot/SnapshotSchedulerImpl.java [hunk 1]
```diff
@@ -264,45 +264,29 @@ protected void deleteExpiredVMSnapshots() {
     @DB
     protected void scheduleSnapshots() {
         String displayTime = DateUtil.displayDateInTimezone(DateUtil.GMT_TIMEZONE, _currentTimestamp);
-        s_logger.debug("Snapshot scheduler.poll is being called at " + displayTime);
+        s_logger.trace(String.format("Snapshot scheduler is being called at [%s].", displayTime));
```
Comment (harikrishna-patnala) [inline server/src/main/java/com/cloud/storage/snapshot/SnapshotSchedulerImpl.java:267]: may be debug is more helpful to know when the scheduler has started !

**Review by harikrishna-patnala — APPROVED** [review]
Code LGTM.

**Review by BryanMLima — APPROVED** [review]

**Review by JoaoJandre — APPROVED** [review]
LGTM

**Comment by DaanHoogland** [comment]
@stephankruggg , i still oppose to unguarded logging with string evaluation in parameters. i have not seen arguments to counter the downside of it yet. So please look at #6973 it implements a uitlity available in log4j2 and other frameworks that will help compromise between our point of views. What do you think?

**Comment by stephankruggg** [comment]
> @stephankruggg , i still oppose to unguarded logging with string evaluation in parameters. i have not seen arguments to counter the downside of it yet. So please look at #6973 it implements a uitlity available in log4j2 and other frameworks that will help compromise between our point of views. What do you think?

Okay, I will take a look at it.

**Comment by stephankruggg** [comment]
@DaanHoogland, sorry for the delay. I took a look at your proposal, and as @GutoVeronezi pointed out (https://github.com/apache/cloudstack/pull/6973#issuecomment-1347285676), we are already working on a Log4j upgrade. Therefore, I do not think applying an extra step is the best option.

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by GutoVeronezi** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Pull request merged**